File: PO/Project/C++/src/guarana.cpp
#include "guarana.h"

guarana::guarana(YX position, world* world_point)
: plant(0, position ,world_point)
{
    avatar = 'u';
}

organizm* guarana::MakeChild(YX new_position)
{
    return (new guarana(new_position, world_point));
}

guarana::~guarana()
{
}

File: PO/Project/JavaProject/src/Plants/Guarana.java
package Plants;

import Abstractions.*;
import java.awt.*;

public class Guarana extends Plant {

    public Guarana(World world,Point position) {
        super(0, position, world);
        this.avatar = 'u';
    }

    @Override
    public boolean rejectAttack(Organism attacker) {
        attacker.setStrength(attacker.getStrength() + 3);
        world.getLogger().logStrengthIncrease(attacker);
        return false;
    }

    @Override
    public Organism makeChild(Point newPosition) {
        return new Guarana(world, newPosition);
    }
}
